cmd/integration-test-server/main.go

```diff
@@ -0,0 +1,192 @@
+	if err != nil {
+		addErrorMessage(fmt.Sprintf("Parse node state failed with error %s", err.Error()))
+		w.Write([]byte(fmt.Sprintf(`{"status":"error", "err message":"%s"}`, err.Error())))
+		return
+	}
+
+	// If no state updated
+	log.Info("Get state from peer", "peer id", state.Id, "Version", state.Version)
+	if lastRecord != nil {
+		checkState(lastRecord, state)
+	}
+
+	lastRecord = state
+
+	w.Write([]byte(`{"status":"ok"}`))
+}
+
+func checkState(oldState, newState *node.NodeState) {
+	if len(oldState.Shards) != len(newState.Shards) {
+		addErrorMessage(fmt.Sprintf("shards count mismatch between two state, new %d, old %d", len(newState.Shards), len(oldState.Shards)))
+		return
+	}
+
+	for _, shardState := range newState.Shards {
+		check := false
+		for _, oldShardState := range oldState.Shards {
+			if shardState.ShardId != oldShardState.ShardId {
+				continue
+			}
+			check = true
+			if shardState.SyncState.PeerCount > 0 {
+				hasConnectedPeer = true
+			}
+
+			if oldShardState.SyncState.SyncProgress < 10000 &&
+				(shardState.SyncState.BlobsSynced <= oldShardState.SyncState.BlobsSynced ||
+					shardState.SyncState.SyncProgress <= oldShardState.SyncState.SyncProgress) {
+				addErrorMessage(fmt.Sprintf("es-node sync progress do not increase in %f minutes, "+
+					"old synced: %d, new synced %d; old progress: %d, new progress: %d", expectedStateRefreshTime.Minutes(), oldShardState.SyncState.BlobsSynced,
+					shardState.SyncState.BlobsSynced, oldShardState.SyncState.SyncProgress, shardState.SyncState.SyncProgress))
+			}
+			if oldShardState.SyncState.FillEmptyProgress < 10000 &&
+				(shardState.SyncState.EmptyFilled < oldShardState.SyncState.EmptyFilled ||
+					shardState.SyncState.FillEmptyProgress < oldShardState.SyncState.FillEmptyProgress) {
+				addErrorMessage(fmt.Sprintf("es-node fill empty progress do not increase in %f minutes, "+
+					"old filled: %d, new filled %d; old progress: %d, new progress: %d", expectedStateRefreshTime.Minutes(), oldShardState.SyncState.EmptyFilled,
+					shardState.SyncState.EmptyFilled, oldShardState.SyncState.FillEmptyProgress, shardState.SyncState.FillEmptyProgress))
+			}
+
+			if oldShardState.SyncState.FillEmptyProgress == 10000 && oldShardState.SyncState.SyncProgress == 10000 &&
+				(shardState.MiningState.MiningPower == 0 || shardState.MiningState.SamplingTime == 0) {
+				addErrorMessage("Mining should be start after sync done.")
+			}
+		}
+		if !check {
+			addErrorMessage(fmt.Sprintf("Shard %d in the new state do not exist in the old state", shardState.ShardId))
+		}
+	}
+}
+
+func checkFinalState(state *node.NodeState) {
+	if state == nil {
+		addErrorMessage("No state submitted during the test")
+		return
+	}
+	if !hasConnectedPeer {
+		addErrorMessage("es-node peer count should larger than 0")
+	}
+
+	log.Info("Final state", "id", state.Id, "version", state.Version)
+	for _, shardState := range state.Shards {
+		if shardState.SyncState.SyncProgress != 10000 {
```

Comment:
Any where to check "Blobs can be downloaded to es-node successfully" && "Synced data is correct."?